### PR TITLE
HPCC-15819 Trace global memory manager and include callbacks

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please review.  Would you prefer it to target 6.0.4 instead??

File: roxie/roxiemem/roxiemem.cpp [hunk 1]
```diff
@@ -4678,6 +4713,8 @@ void CHugeHeap::expandHeap(void * original, memsize_t copysize, memsize_t oldcap
     unsigned newPages = PAGES(newsize + HugeHeaplet::dataOffset(), HEAP_ALIGNMENT_SIZE);
     unsigned oldPages = PAGES(oldcapacity + HugeHeaplet::dataOffset(), HEAP_ALIGNMENT_SIZE);
     void *oldbase =  (void *) ((memsize_t) original & HEAP_ALIGNMENT_MASK);
+    HugeHeaplet * oldHeap = (HugeHeaplet * )oldbase;
```
Comment (jakesmith) [inline roxie/roxiemem/roxiemem.cpp]: v.trivial: extra space in (HugeHeaplet * )

**Comment by jakesmith** [comment]
@ghalliday - looks good. Targeting master/6.2 is fine.

**Comment by ghalliday** [comment]
@jakesmith removed space, and made the is-this-the-correct-heap check slightly cleaner

**Comment by jakesmith** [comment]
@ghalliday - looks good to merge.

**Comment by ghalliday** [comment]
@richardkchapman please merge

**Comment by jakesmith** [comment]
I think this should go into 6.0.4 if not 6.0.2 (which we're probably about to build another RC for)
@ghalliday @richardkchapman

**Comment by richardkchapman** [comment]
@ghalliday Please retarget to 6.0.2

**Pull request closed without merging**